smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -86,8 +76,33 @@ public ApplicationProtocol getApplicationProtocol() {
      */
     protected abstract String getDocumentContentType();
 
+    /**
+     * Generates serialization functions for shapes in the passed set. These functions
+     * should return a value that can then be serialized by the implementation of
+     * {@code serializeDocument}. The {@link DocumentShapeSerVisitor} and {@link DocumentMemberSerVisitor}
+     * are provided to reduce the effort of this implementation.
+     *
+     * @param context The generation context.
+     * @param shapes The shapes to generate serialization for.
+     */
+    protected abstract void generateDocumentShapeSerializers(GenerationContext context, Set<Shape> shapes);
+
+    /**
+     * Generates deserialization functions for shapes in the passed set. These functions
+     * should return a value that can then be deserialized by the implementation of
+     * {@code deserializeDocument}. The {@link DocumentShapeDeserVisitor} and
+     * {@link DocumentMemberDeserVisitor} are provided to reduce the effort of this implementation.
+     *
+     * @param context The generation context.
+     * @param shapes The shapes to generate deserialization for.
+     */
+    protected abstract void generateDocumentShapeDeserializers(GenerationContext context, Set<Shape> shapes);
```

Comment:
Should this and generateDocumentShapeSerializers just become a single method, `generateDocumentShapeSerdeFunctions`? It seems like them as separate methods doesn't actually help anything, does it? Or are the provided Sets totally different?

Reply:
The sets themselves are likely to be quite distinct. The most obvious example is for errors, but an API may consume shapes that aren't given as output, and vice versa, and generating functions for those is unused overhead.